Sources/KlaviyoForms/InAppForms/IAFPresentationManager.swift

```diff
@@ -118,26 +119,44 @@ class IAFPresentationManager {
         viewController?.modalPresentationStyle = .overCurrentContext
     }
 
-    // MARK: - Form Event Subscription
+    // MARK: - Form Lifecycle Listener Setup
 
-    private func listenForFormEvents() {
+    private func setupFormLifecycleListener() {
         guard let viewModel else { return }
 
+        if #available(iOS 14.0, *) {
+            Logger.webViewLogger.info("👂 Starting to listen for form lifecycle events (BEFORE handshake)")
+        }
+
+        // Start listening for form lifecycle events before handshake to avoid missing any events
+        formEventTask = Task { [weak self] in
+            guard let self else { return }
+            for await event in viewModel.formLifecycleStream {
+                self.handleFormEvent(event)
+            }
+        }
```

Comment:
This was once code change that AI suggested that was a good one. Earlier we were first establishing the handshake and then listening to the data emitted by JS like `present`, `dismiss` and `abort`. The change here is we first listen to the data emitted by JS and then establish the handshake to avoid any loss of data coming from the web view.

Reply:
yeah that makes sense, no need to wait from what i can tell.